### DISCUSSION TITLE
A suggestion on a new application

### DISCUSSION THREAD

**Discussion by mmcheraghi** [stnolting/neorv32#1106] [Ideas]
Hi @stnolting 
I think it's possible to use Neorv32 as a Modbus RTU server and communicate with it through some modbus master simulators in windows.

It only requires an RS232 connection to laptop serial port and a Modbus RTU software on Neorv32.

I think Its a useful application. An one could learn modbus protocol with it.

**Comment by stnolting** [comment]
Hey @mahdi259.

As far as I know Modbus just requires a standard UART - so that should be no problem. So yes, this would be a cool application!

Btw, I am using the processor as a controller/gateway for "home automation" (controlling lights, reading sensors, etc.). But instead of using Modbus I am using the 1-wire interface. It makes wiring pretty easy and you can get off-the-shelves components.

Anyway, Modbus sounds great! Keep us updated! :wink:

**Reply by mmcheraghi** [reply]
Ok Thanks
If I find a good modbus server code I will share that here.
